### ISSUE TITLE
Add support for Chromium viewport configuration for screenshots

### ISSUE THREAD

**Issue by gg-gaborginter** [gotenberg/gotenberg#786]
While for generating PDFs, there is the option to manipulate the sizing of the end result through e.g. the "paperWidth" and "paperHeight" request attributes, in case of screenshots, the feature seems to be missing yet from the current route API.

I recommend the following additional configuration options for screenshots, which are fully supported by Chromium:
- `captureBeyondViewport`: its current hard-coded value is `true` in spite of the Chromium API default being `false`; optional, the default value should stay `true` for backward compatibility reasons
- viewportWidth: similarly to the HTML viewport meta tag, an integer value, designating number of pixels; optional, its default value would be `-1`, in which case the viewport restriction is not passed over to Chromium
- viewportHeight: similarly to the HTML viewport meta tag, an integer value, designating number of pixels; optional, its default value would be `-1`, in which case the viewport restriction is not passed over to Chromium

**Comment by gulien** [comment]
Hey @gg-gaborginter,

I was playing with `viewportWidth` and `viewportHeight`, and I was not able to make them work. 
For instance, increasing the value of `viewportWidth` did not make the website CSS "adapt", but instead it kept the same width while capturing more "background". `viewportHeight` was working OK, meaning I was able to take screenshots of a portion of the site.

Also, keep in mind that screenshots are not a first class citizen, but merely a tool to make a sort of "preview" of the PDF converssion.

**Comment by gg-gaborginter** [comment]
Hi @gulien,

We are at the moment using a self-built, similar setup for both purposes: PDF and screenshot generation. With screenshots, as the window size can't be influenced anyhow else on the Chromium API, we set the viewport sizes instead and restricted the width of the page body through CSS.

I made some tests with the HTML viewport meta tag, by fixing the width and height explicitly, but without the viewport setting on the Chromium page instance, the screenshot was made of the whole window size (lots of white space all around the desired content).
 
I understand that the screenshot feature is not a first-class citizen. I merely would like to have the option to control some of the settings that Chromium offers. I'm willing to contribute, so I created a pull request, as well: https://github.com/gotenberg/gotenberg/pull/787
